lib/doctest.js

```diff
@@ -39,36 +40,47 @@ module.exports = function(path, options) {
   if (options.module != null &&
       options.module !== 'amd' &&
       options.module !== 'commonjs') {
-    throw new Error ('Invalid module `' + options.module + "'");
+    return Promise.reject (new Error (
+      'Invalid module `' + options.module + "'"
+    ));
   }
   if (options.type != null &&
       options.type !== 'coffee' &&
       options.type !== 'js') {
-    throw new Error ('Invalid type `' + options.type + "'");
+    return Promise.reject (new Error (
+      'Invalid type `' + options.type + "'"
+    ));
+  }
+
+  var type = options.type == null ? inferType (path) : options.type;
+
+  if (type == null) {
```

Comment:
I'd like to remove the empty line preceding the `if`, as I consider the assignment and the check part of the same thought.